libsolidity/parsing/Parser.cpp

```diff
@@ -322,11 +322,15 @@ StateMutability Parser::parseStateMutability(Token::Value _token)
 	StateMutability stateMutability(StateMutability::NonPayable);
 	if (_token == Token::Payable)
 		stateMutability = StateMutability::Payable;
-	// FIXME: constant should be removed at the next breaking release
-	else if (_token == Token::View || _token == Token::Constant)
+	else if (_token == Token::View)
 		stateMutability = StateMutability::View;
 	else if (_token == Token::Pure)
 		stateMutability = StateMutability::Pure;
+	else if (_token == Token::Constant)
+		fatalParserError(
```

Comment:
If you make this fatal, the users will only get a single error on the first `constant`. You can generate a regular parse error here and continue like in `View`.